Simplified/NYPLBook.m

```diff
@@ -329,9 +226,151 @@ - (instancetype)initWithDictionary:(NSDictionary *)dictionary
 {
   self = [super init];
   if(!self) return nil;
-  
-  self.acquisition = [[NYPLBookAcquisition alloc] initWithDictionary:dictionary[AcquisitionKey]];
-  if(!self.acquisition) return nil;
+
+  // This is not present in older versions of serialized books.
+  NSArray *const acquisitionsArray = dictionary[AcquisitionsKey];
+  if (acquisitionsArray) {
+    assert([acquisitionsArray isKindOfClass:[NSArray class]]);
+
+    NSMutableArray<NYPLOPDSAcquisition *> *const mutableAcqusitions =
+      [NSMutableArray arrayWithCapacity:acquisitionsArray.count];
+
+    for (NSDictionary *const acquisitionDictionary in acquisitionsArray) {
+      assert([acquisitionDictionary isKindOfClass:[NSDictionary class]]);
+
+      NYPLOPDSAcquisition *const acquisition = [NYPLOPDSAcquisition acquisitionWithDictionary:acquisitionDictionary];
+      assert(acquisition);
+
+      [mutableAcqusitions addObject:acquisition];
```

Comment:
if the assertion. fails in non-debug mode this could try and append `nil` here and 💣 , should we also check its presence in the app?

Reply:
We do not define `NDEBUG` for release builds so assertions made with `assert` are always checked. In any case, the assertion here is just to check that my logic was correct: The assertion should _always_ be true, and if it isn't, it indicates a bug in the code that needs to be fixed. As such, I'd actually prefer the app crash so we can see it and fix it as opposed to just quietly bailing out of the method and logging something that'll never be noticed. I hope that makes sense!